tests/spread/extensions/env-injector/task.yaml

```diff
@@ -0,0 +1,84 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  cd "$SNAP_DIR"
+  snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the hello command is present
+  [ -f squashfs-root/usr/local/bin/hello ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous
+
+  echo "[env-injector] Creating global envfile"
+  # Create global envfile
+  echo 'HELLO_WORLD=Hello World' >> envfile.env
+
+  echo "[env-injector] Creating app-specific envfile"
+  # Create app-specific envfile
+  echo 'SCOPED=Scoped' >> appenv.env
+
+  echo "[env-injector] Setting global env variable"
+  # Set env vars: Global
+  snap set env-injector-hello env.global="World"
```

Comment:
What was the decision on a configure hook?  If I understand correctly, this will fail because there is no configure hook.

Reply:
Sorry, you're right, this is missing a configure hook. 

We're not including it in the extension because the user may already have one. However, if they don't, snapd will notify them that a configure hook is required.

Reply:
The design envisioned that the extension would add an empty configure hook when it wasn't present. But this wasn't straight forward because the configure hook could be added from the project source or during the parts build. Detecting the lack of a configure hook from source and a part is possible, as suggestion [here](https://matrix.to/#/!GGqzbFAUQprdPgYYCM:ubuntu.com/$3upkz8HmiNl-hcw2il79j9GWEFORRQ7XPQ_1EtL4jdc?via=ubuntu.com&via=matrix.org&via=kde.org) by first checking `$CRAFT_PROJECT_DIR/snap/hooks/configure` and also adding a part that runs `after` all others to check if no other part has primed it (e.g. a compiled hook or one added from a different source).

In the end, we decided to leave this out for this version of the implementation and instead document that the extension should be used with a snap that has a configure hook. (In the meantime, we could look into why it is even mandatory to have a configure hook for a snap to accept configurations, where many snaps work around it by adding an empty script that does no config validations).